petab/core.py

```diff
@@ -149,28 +154,33 @@ def get_constant_parameters(self):
         to optimization, no sensitivities w.r.t. these parameters are
         required).
         """
+        warnings.warn("This function will be removed in future releases. ",
```

Comment:
why?

Reply:
To stop abusing sbml parameter `constant` attributes. We started using that for AMICI model generation. For that it needs to be decided elsewhere what should be considered a constant/fixed parameter. If we want to keep something similar here, it should return all parameters with estimate=False and those which are overridden with such a parameter for all conditions. But in the current form, this function is obsolete in my opinion.